cmd/e3db/main.go

```diff
@@ -590,6 +593,79 @@ func cmdPolicyIncoming(cmd *cli.Cmd) {
 	}
 }
 
+func cmdRegister(cmd *cli.Cmd) {
+	apiBaseURL := cmd.String(cli.StringOpt{
+		Name:      "api",
+		Desc:      "e3db api base url",
+		Value:     "",
+		HideValue: true,
+	})
+
+	email := cmd.String(cli.StringArg{
+		Name:      "EMAIL",
+		Desc:      "client e-mail address",
+		Value:     "",
+		HideValue: true,
+	})
+
+	token := cmd.String(cli.StringArg{
+		Name:      "TOKEN",
+		Desc:      "registration token from the InnoVault admin console",
+		Value:     "",
+		HideValue: false,
+	})
+
+	cmd.Action = func() {
+		// Preflight check for existing configuration file to prevent a later
+		// failure writing the file (since we use O_EXCL) after registration.
+		if e3db.ProfileExists(*options.Profile) {
+			var name string
+			if *options.Profile != "" {
+				name = *options.Profile
+			} else {
+				name = "(default)"
+			}
+
+			dieFmt("register: profile %s already registered", name)
+		}
+
+		// minimally validate that email looks like an email address
+		_, err := mail.ParseAddress(*email)
+		if err != nil {
+			dieErr(err)
+		}
+
+		pub, priv, err := box.GenerateKey(rand.Reader)
+		if err != nil {
+			dieErr(err)
+		}
+
+		publicKey := e3db.ClientKey{Curve25519: base64.RawURLEncoding.EncodeToString(pub[:])}
```

Comment:
Should we provide some kind of helper to generate the keys? It feels a bit like an interface fail that the caller of `RegisterClient` needs to import NaCL to do the key generation.